README.md

```diff
@@ -723,6 +724,45 @@ consumer.each_message do |message|
 end
 ```
 
+#### Customizing Partition Assignment Strategy
+
+In some cases, you might want to assign more partitions to some consumers. For example, in applications inserting some records to a database, the consumers running on hosts nearby the database can process more messages than the consumers running on other hosts.
+You can implement a custom assignment strategy and use it by passing a proc object that create the strategy as the argument `assignment_strategy_builder` like below:
+
+```ruby
+class CustomAssignmentStrategy
+  def initialize(cluster, user_data)
+    @cluster = cluster
+    @user_data = user_data
+  end
+
+  # @return [String]
+  def protocol_name
+    ...
+  end
+
+  # @return [String, nil]
+  def user_data
+    @user_data
+  end
+
+  # Assign the topic partitions to the group members.
+  #
+  # @param members [Hash<String, Protocol::JoinGroupResponse::Metadata>] a hash
+  #   mapping member ids to metadata
+  # @param topics [Array<String>] topics
+  # @return [Hash<String, Protocol::MemberAssignment>] a hash mapping member
+  #   ids to assignments.
+  def assign(members:, topics:)
+    ...
+  end
+end
+
+assignment_strategy_builder = ->(cluster) do
+  CustomAssignmentStrategy.new(cluster, "some-host-information")
```

Comment:
Not sure it’s necessary to expose this as a full builder pattern – maybe just a simpler callback, e.g. `(cluster, user_data, members, topics) -> assignments`?

Reply:
The strategy must also implement `#protocol_name` and `#user_data`, so I think it is difficult to replace it with a simpler callback.

Reply:
:+1: